tests/test_properties.py

```diff
@@ -40,6 +40,108 @@ def eval(self, x):
         return self.a * x + self.b
 
 
+def g(x):
+    """Non-linear model, Example 5.1 from Evensen 2019"""
+    x1 = x[0]
+    return x1 + 0.2 * x1**3
+
+
+@pytest.mark.parametrize("N", [100, 200])
+def test_that_projection_is_better_for_nonlinear_forward_model_big_N_small_p(N):
+    # For non-linear forward model, g, and m<N,
+    # Eq.27 should provide a better update than Eq. 28.
+    # "Better" in terms of a better optimum, given by loss in Eq. 10
+    # We here solve using both Equations 27 and 28 (with/without projection)
+    # and then evaluate the loss function.
+
+    def loss_function(xj, xj_prior, dj, Cxx, Cdd, g):
+        """Equation 10 in Evensen 2019"""
+        return 0.5 * (
+            (xj - xj_prior).T @ np.linalg.inv(Cxx) @ (xj - xj_prior)
+            + (g(xj) - dj).T @ np.linalg.inv(Cdd) @ (g(xj) - dj)
+        )
+
+    N = 100
+    m = 1
+    x_true = np.array([-1.0])
+
+    # sample parameters from prior
+    x_sd = 1.0
+    prior_bias = 0.5
+    X_prior = np.array(
```

Comment:
Don't think that `np.array` nor the `reshape` is necessary as `np.random.normal` with `size=(m, N)` should return an array of size `(m, N)`.

Also, might as well use `rng = np.random.default_rng()` as random number generator as it is recommended by numpy.

Reply:
Thanks, I will try this. 👍